api/java/org/openyolo/api/internal/CredentialRetrieveActivity.java

```diff
@@ -106,23 +113,27 @@ public void onResponse(long queryId, List<QueryResponse> queryResponses) {
 
                 protoResponses.put(queryResponse.responderPackage, response);
 
-                if (response.getRetrieveIntent().isEmpty()) {
-                    continue;
-                }
 
-                Intent retrieveIntent;
-                try {
-                    retrieveIntent =
-                            IntentUtil.fromBytes(response.getRetrieveIntent().toByteArray());
-                } catch (BadParcelableException ex) {
-                    Log.w(LOG_TAG, "Failed to parse intent from bytes");
+                // a provider indicates that it has credentials via the credentials_available field
+                // on the response proto. Prior 0.3.0, the provider would directly return an intent.
+                // For backwards compatibility this is currently inspected as a fallback for the
+                // absence of the credentials_available field, but the Intent itself is no longer
+                // used.
+                // TODO: remove backwards compatibility with retrieve_intent after 0.3.0
+                if (!response.getCredentialsAvailable()
+                        && !response.getRetrieveIntent().isEmpty()) {
```

Comment:
`&& !response.getRetrieveIntent().isEmpty()`
=> `&& response.getRetrieveIntent().isEmpty()`, no?

Reply:
Removed as the code has been reordered to fit your request for using the old intent as a fallback.